gtsfm/averaging/rotation/shonan.py

```diff
@@ -35,6 +37,36 @@
 _DEFAULT_TWO_VIEW_ROTATION_SIGMA = 1.0
 
 
+def random_rotation() -> Rot3:
+    """Sample a random rotation by generating a sample from the 4d unit sphere."""
+    q = np.random.randn(4) * 0.03
+    # make unit-length quaternion
+    q /= np.linalg.norm(q)
+    qw, qx, qy, qz = q
+    R = Rot3(qw, qx, qy, qz)
+    return R
+
+
+def initialize_global_rotations_using_mst(num_images: int, i2Ri1_dict: Dict[Tuple[int, int], Rot3]) -> List[Rot3]:
```

Comment:
Thanks for adding!

could we put this function in a small adjacent helper file? This file is getting a bit big. Can we add a test in a simple 3 or 4 edge example?

Reply:
Marking as resolved, thanks for splitting off into the other file.